src/tainting/Dataflow_tainting.ml

```diff
@@ -1435,8 +1435,17 @@ let check_function_call_arguments env args =
  * report the finding too (by side effect). *)
 let check_tainted_instr env instr : Taints.t * Lval_env.t =
   let check_expr env = check_tainted_expr env in
-  let check_instr = function
-    | Assign (_, e) -> check_expr env e
+  let rec check_instr = function
```

Comment:
We would need a test case and a changelog too.

Reply:
Fixed.